moria.6
.TH MORIA 6
.UC 4
.SH NAME
moria \- Dungeon Adventure
.SH SYNOPSIS
.B /usr/games/moria
[
.B \-ros
]
[
.I save_file
]
.SH DESCRIPTION
.PP
The game of
.I moria
is a single player dungeon simulation.   A
player  may  choose  from  a  number  of  races  and classes when
creating their character, and then 'run' that  character  over  a
period of days, weeks, even months; attempting to win the game by
defeating the Balrog which lurks in the deeper levels.
.PP
The player will begin his adventure on the town level  where
he  may  acquire supplies, weapons, armor, and magical devices by
bartering with various shop  owners.   After  preparing  for  his
adventure,  the  player  can  descend  into the dungeons of
.I moria
where fantastic adventures await his coming!
.PP
To get started you really only need to know two commands.  The command
.B ?
will give you a list of the available commands (below).
The
.B /
command will identify the objects you see on the screen.
Before beginning your first adventure, you should also read
.I The Dungeons of MORIA
document carefully.  The game of
.I moria
is a complicated game, and
will require a dedicated player to win.
.PP
The
.B \-r
switch will make the game accept rogue like commands.
.PP
The
.B \-o
switch will make the game accept original commands (this is the default).
.PP
The
.B \-s
switch will print the scores and then quit.
.PP
If
.I save_file
is specified,
.I moria
will be restored from the specified saved game file.
.SH "Original Mode Commands"
.nf
B <Dir> Bash (object/creature)|  q        Quaff a potion.
C       Display character.    |  r        Read a scroll.
D <Dir> Disarm a trap/chest.  |  s        Search for hidden doors.
E       Eat some food.        |  t        Take off an item.
F       Fill lamp with oil.   |  u        Use a staff.
L       Current location.     |  v        Version and credits.
P       Print map.            |  w        Wear/Wield an item.
R       Rest for a period.    |  x        Exchange weapon.
S       Search Mode.          |  /        Identify a character.
T <Dir> Tunnel.               |  ?        Display this panel.
a       Aim and fire a wand.  |
b       Browse a book.        |  ^M       Repeat the last message.
c <Dir> Close a door.         |  ^R       Redraw the screen.
d       Drop an item.         |  ^K       Quit the game.
e       Equipment list.       |  ^X       Save character and quit.
f       Fire/Throw an item.   |   $       Shell out of game.
i       Inventory list.       |  < Go up an up-staircase.
j <Dir> Jam a door with spike.|  > Go down a down-staircase
l <Dir> Look given direction. |  . <Dir>  Move in direction.
m       Cast a magic spell.   |  Movement: 7  8  9
o <Dir> Open a door/chest.    |            4     6    5 == Rest
p       Read a prayer.        |            1  2  3
.fi
.bp
.SH "Rogue Like Commands"
.nf
f <Dir> Bash (object/creature)|  q        Quaff a potion.
C       Display character.    |  r        Read a scroll.
D <Dir> Disarm a trap/chest.  |  s        Search for hidden doors.
E       Eat some food.        |  T        Take off an item.
F       Fill lamp with oil.   |  Z        Use a staff.
W       Current location.     |  v        Version and credits.
M       Print map.            |  w        Wear/Wield an item.
R       Rest for a period.    |  X        Exchange weapon.
#       Search Mode.          |  /        Identify a character.
CTRL <dir> Tunnel             |  ?        Display this panel.
z       Aim and fire a wand.  |
P       Browse a book.        |  ^P       Repeat the last message.
c <Dir> Close a door.         |  ^R       Redraw the screen.
d       Drop an item.         |  Q        Quit the game.
e       Equipment list.       |  ^X       Save character and quit.
t       Fire/Throw an item.   |   !       Shell out of game.
i       Inventory list.       |  < Go up an up-staircase.
S <Dir> Jam a door with spike.|  > Go down a down-staircase.
x <Dir> Look given direction. |  SHIFT <Dir>  Move in direction.
m       Cast a magic spell.   |  Movement: y  k  u
o <Dir> Open a door/chest.    |            h     l    . == Rest
p       Read a prayer.        |            b  j  n
.fi
.SH AUTHORS
Robert Alan Koeneke / University of Oklahoma
.br
Jimmey Wayne Todd   / University of Oklahoma
.SH UNIX Port
James E. Wilson     / UC Berkeley
.SH Manual Page
Roger S. Southwick  / Tektronix, Inc.
.SH BUGS
.PP
The SYS V version does not lock the scoreboard, if two processes try
to modify it at the same time, it may become corrupted.

Wearing an amulet that would raise a stat over 18/100 actually only raises
the stat to 18/100.  When the amulet is removed, the stat drops to a value
below its original value.

Probably infinite (although countably infinite).


CHANGES
::::::: 1990 ::::::::
------------ 1/1
creatures.c: in mon_cast_spell(), must update_mon() to light monsters before
	they cast a spell
------------- 1/2
moria1.c: inven_command(), could not wear item which replaced equip item if
	inven full, only fail if inven_check_num fails for torches
misc2.c: change spacing for stats on change_character page, was printing in
	last column for max stat
moria2.c: tunnel(), if no shovel/pick, divide total tabil chance by two
monsters.c: change eye-sight and sleep values for consistency, as noted by
	djgrabin
help.c: change the monster recall so that it lists high level monsters first
creature.c: when monster bashes down a door, call disturb() to halt resting
io.c: fix msg_print() so that -more- is always printed on screen, fix
	put_buffer() so that does not fail if col argument greater than 79/80
misc1.c: change chance for ego weapons/missiles to 3*special/2 instead of
	just special, needed because treasure distribution change has made
	ego weapons not as common as before
--------- 1/11
monsters.c: let nether wraith phase, set Black/Red/Multi ancient dragons
	to a sleep value of 700
creature.c: increase r_attacks if either notice true or r_attacks is
	nonzero (i.e. it has been noticed before)
moria2.c: move_char(), when step back after setting off a falling rock trap,
	check to see if have stepped back onto another trap
moria1.c: inven_command(), when restart it after taking off an item, don't
	print the 'have to drop something' message because that is confusing
misc2.c: modify_stat(), could overflow 118 when increasing stats, add extra
	case to enforce max of 118
--------- 1/12
all: apply william setzer's objdes() diffs, 500K patch file, change version
	number to 5.0.10 and distribute
desc.c, misc2.c, store1.c: remove several obsolete uses of index(),
	insert_num() no longer used anywhere
constant.h, desc.c: change ID_TRIED,ID_KNOWN1 to OD_TRIED,OD_KNOWN1
desc.c: eliminate titles[], rename title_space[] to titles[]
all: fixed lint errors
all: make sure that all objdes() strings are bigvtype
desc.c: fix printing of bows/arrows/picks, needed new p1_use define, and
	rearrange code in objdes() switch statement
treasur1.c, treasur2.c: remove -100 tohit from books, was being printed, and
	is silly anyways
constant.h, externs.h, save.c, treasur2.c: change object_ident[] from
	MAX_OBJECTS to OBJECT_IDENT_SIZE = 7*64, see object_offset() in desc.c
	to see why this is necessary
---------- 1/13
externs.h, misc2.c, store1.c: deleted obsolete references to treasure_type
desc.c: always print toac field if known2 and non-negative
config.h, io.c, misc1.c: eliminated BUGGY_CURSES define, caused too much
	trouble, just assume that can never print in last column of screen
constant.h, desc.c, misc1.c, moria2.c, spells.c, treasur1.c, types.h:
	created special_names table to hold all possible name2 strings,
	name2 changed to an int8u which holds index to special_names table
externs.h, main.c, treasure1.c: sort_objects() function deleted, sorting now
	done by init_t_level using O(n) bin sort, into sorted_objects[] array
files.c, misc2.c: object_list[] references that assumed it was sorted now
	indirect through sorted_objects[] array
main.c, variable.c: player_init indexes now point into object_list instead
	of inventory_init
------- 1/14
constant.h, externs.h, main.c, store1.c, treasur1.c, treasur2.c, variable.c:
	deleted inventory_init array, stores objects now in object_list,
	MAX_OBJECT adjusted, MAX_DUNGEON_OBJ defined, store_choice[] fixed
all: put all objects in object_list, trap_list, door_list, store_door,
	gold_list, up_stair, down_stair, rubble, much, blank_treasure,
	scare_monster are not separate variables anymore
all: invcopy now takes an object_list index as second parameter, name field
	deleted from inven_type and fields rearranged
----------- 1/15
store1.c: search_list() deleted, use i_ptr->index instead to get raw item cost
save.c: obsolete treasure_type references deleted, save files work again
creature.c, magic.c, misc2.c, prayer.c, save.c, wands.c: fixed long/short
	problems with shift, mostly change (1 << i) to (1L << i)
all: add new register declaration, check indentation, fix 32/16bit int problems
constant.h: change version number to 5.0.11
------------- 1/17
spells.c: aggravate was not initiallized in aggravate_monster()
monsters.c: novice rogue now evil like other rogues
misc2.c: 2 "... learn ... new spells ..." messages changed to print prayers
	instead for priests
moria2.c: drop_throw(), "The %s hits %s" changed to "The %s hits the %s."
many: fixed problems with setting of free_turn_flag
----------- 1/19
misc2.c: fix inven_carry(), to stack store/dungeon items properly
desc.c: objdes(), set p1_use = IGNORED for food and potions, changed
	is_a_vowel(tmp_val[1]) to [2]
moria2.c: "exceed weight limit?" message now prints item name also
desc.c, spells.c, staffs.c, wands.c: empty wands are not incribed empty if
	known2, empty removed when wands are recharged
desc.c: put spaces between inscribed words
dungeon.c: always inscribe majik when character notices this, not just for
	inventory items as that helps priests more than fighters
store1.c: sell_price(), succeed only if both item->cost and item_value() are
	positive
creature.c: don't increase a number of attacks when monster repelled in
	make_attack(), i.e. no damage done so shouldn't learn about damage

mac/moria.r
#include "Types.r"

include "ScrnMgr.rsrc";

type 'MRIA' as 'STR ';

type 'CNFG' (256 : 257) {
	longint;
};

type 'TEXT' {
	string;
};

resource 'MRIA' (0) {
	"Moria 5.2.1 for the Macintosh: Implementation 2.0b1"
};

resource 'BNDL' (128, purgeable) {
	'MRIA',
	0,
	{ /* array TypeArray: 2 elements */
		/* [1] */
		'ICN#',
		{ /* array IDArray: 3 elements */
			/* [1] Application */
			0, 128,
			/* [2] Save file */
			1, 129,
			/* [3] Aux files */
			2, 130
		},
		/* [2] */
		'FREF',
		{ /* array IDArray: 5 elements */
			/* [1] Appilication */
			0, 128,
			/* [2] Save file */
			1, 129,
			/* [3] Info files */
			2, 130,
			/* [4] Score file */
			3, 131,
			/* [5] Config file */
			4, 132
		}
	}
};

resource 'FREF' (128, "Application", purgeable) {
	'APPL',
	0,
	""
};

resource 'FREF' (129, "Save Files", purgeable) {
	'SAVE',
	1,
	""
};

resource 'FREF' (130, "Info Files", purgeable) {
	'TEXT',
	2,
	""
};

resource 'FREF' (131, "Score File", purgeable) {
	'SCOR',
	2,
	""
};

resource 'FREF' (132, "Config File", purgeable) {
	'CNFG',
	2,
	""
};

resource 'ICON' (128, "Application", purgeable) {
	$"00 01 00 00 00 02 80 00 00 04 40 00 00 08 20 00"
	$"00 10 10 00 00 22 08 00 00 45 04 00 00 88 82 00"
	$"01 05 15 00 02 02 A8 80 04 40 40 40 08 A8 80 20"
	$"11 15 00 10 24 8A 00 28 40 51 3F 44 91 20 C0 8A"
	$"40 00 80 51 24 41 30 22 10 01 C8 14 09 1E 7F 8F"
	$"04 02 30 07 02 41 00 07 01 00 80 07 00 80 60 07"
	$"00 40 1F E7 00 20 02 1F 00 10 04 07 00 08 08 00"
	$"00 04 10 00 00 02 20 00 00 01 40 00 00 00 80"
};

resource 'ICON' (256, "Command Set", purgeable) {
	$"00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00"
	$"00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00"
	$"00 00 00 00 7F FF FF FE 80 00 00 01 9A AA AA A9"
	$"80 00 00 01 95 55 55 59 80 00 00 01 9A AA AA A9"
	$"80 00 00 01 85 7F FF 51 80 00 00 01 80 00 00 01"
	$"7F FF FF FE"
};

resource 'ICON' (257, "TEXT Editor", purgeable) {
	$"0F FF FE 00 08 00 03 00 08 00 02 80 08 00 02 40"
	$"08 00 02 20 08 00 02 10 08 00 03 F8 08 00 00 08"
	$"08 00 00 08 08 07 C0 08 08 0F E0 08 08 18 30 08"
	$"08 30 18 08 08 20 18 08 08 00 18 08 08 00 30 08"
	$"08 00 60 08 08 00 C0 08 08 01 80 08 08 01 80 08"
	$"08 01 80 08 08 01 80 08 08 00 00 08 08 01 80 08"
	$"08 03 C0 08 08 01 80 08 08 00 00 08 08 00 00 08"
	$"08 00 00 08 08 00 00 08 08 00 00 08 0F FF FF F8"
};

resource 'ICN#' (128, "Application", purgeable) {
	{ /* array: 2 elements */
		/* [1] */
		$"00 01 00 00 00 02 80 00 00 04 40 00 00 08 20 00"
		$"00 10 10 00 00 22 08 00 00 45 04 00 00 88 82 00"
		$"01 05 15 00 02 02 A8 80 04 40 40 40 08 A8 80 20"
		$"11 15 00 10 24 8A 00 28 40 51 3F 44 91 20 C0 8A"
		$"40 00 80 51 24 41 30 22 10 01 C8 14 09 1E 7F 8F"
		$"04 02 30 07 02 41 00 07 01 00 80 07 00 80 60 07"
		$"00 40 1F E7 00 20 02 1F 00 10 04 07 00 08 08 00"
		$"00 04 10 00 00 02 20 00 00 01 40 00 00 00 80",
		/* [2] */
		$"00 01 00 00 00 03 80 00 00 07 C0 00 00 0F E0 00"
		$"00 1F F0 00 00 3F F8 00 00 7F FC 00 00 FF FE 00"
		$"01 FF FF 00 03 FF FF 80 07 FF FF C0 0F FF FF E0"
		$"1F FF FF F0 3F FF FF F8 7F FF FF FC FF FF FF FE"
		$"7F FF FF FF 3F FF FF FE 1F FF FF FC 0F FF FF FF"
		$"07 FF FF FF 03 FF FF FF 01 FF FF FF 00 FF FF FF"
		$"00 7F FF FF 00 3F FE 1F 00 1F FC 07 00 0F F8 00"
		$"00 07 F0 00 00 03 E0 00 00 01 C0 00 00 00 80"
	}
};

resource 'ICN#' (129, "Save Files", purgeable) {
	{ /* array: 2 elements */
		/* [1] */
		$"0F FF FE 00 08 00 03 00 08 00 02 80 08 00 02 40"
		$"08 00 02 20 08 00 02 10 08 00 03 F8 08 00 00 08"
		$"08 00 00 08 08 7F FE 08 08 FF FF 08 09 80 01 88"
		$"09 9F F9 88 09 B0 09 88 09 B1 29 88 09 92 A9 88"
		$"09 91 29 88 09 B0 49 88 09 B0 09 88 09 9F F9 88"
		$"09 80 01 88 09 FF FF 88 09 FF FF 88 09 80 01 88"
		$"08 00 00 08 09 12 64 88 09 B5 55 48 09 55 65 C8"
		$"09 15 55 48 09 12 55 48 08 00 00 08 0F FF FF F8",
		/* [2] */
		$"0F FF FE 00 0F FF FF 00 0F FF FF 80 0F FF FF C0"
		$"0F FF FF E0 0F FF FF F0 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
	}
};

resource 'ICN#' (130, "Aux Files", purgeable) {
	{ /* array: 2 elements */
		/* [1] */
		$"0F FF FE 00 08 00 03 00 08 00 02 80 09 FB AE 40"
		$"08 00 02 20 08 00 02 10 09 F7 7B F8 08 00 00 08"
		$"08 00 00 08 09 BB ED C8 08 00 00 08 08 00 00 08"
		$"09 F7 B7 C8 08 00 00 08 08 00 00 08 09 B5 FD C8"
		$"08 00 00 08 08 00 00 08 09 77 B6 C8 08 00 00 08"
		$"08 00 00 08 09 EB B7 48 08 00 00 08 08 00 00 08"
		$"08 00 00 08 09 12 64 88 09 B5 55 48 09 55 65 C8"
		$"09 15 55 48 09 12 55 48 08 00 00 08 0F FF FF F8",
		/* [2] */
		$"0F FF FE 00 0F FF FF 00 0F FF FF 80 0F FF FF C0"
		$"0F FF FF E0 0F FF FF F0 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
		$"0F FF FF F8 0F FF FF F8 0F FF FF F8 0F FF FF F8"
	}
};

resource 'DLOG' (128, "About", purgeable) {
	{46, 26, 304, 274},
	dBoxProc,
	invisible,
	noGoAway,
	0x0,
	128,
	""
};

resource 'DITL' (128, "About", purgeable) {
	{	/* array DITLarray: 4 elements */
		/* [1] */
		{224, 192, 247, 235},
		Button {
			enabled,
			"OK"
		},
		/* [2] */
		{232, 224, 241, 233},
		UserItem {
			disabled
		},
		/* [3] */
		{16, 8, 248, 184},
		StaticText {
			disabled,
			"Moria for the Macintosh\n"
			"Version 5.2.1\n"
			"Implementation 2.0b1\n\n\n"
			"Programmers:\n"
			"Robert Alan Koeneke\n"
			"Jimmey Wayne Todd\n\n"
			"UNIX Port:\n"
			"James E. Wilson\n\n"
			"Macintosh Port:\n"
			"Curtis W. McCauley"
		},
		/* [4] */
		{16, 195, 48, 227},
		Icon {
			disabled,
			128
		}
	}
};

resource 'STR ' (128, "About Title", purgeable) {
	"About Moria\311"
};

resource 'CNFG' (256, "Command Set", purgeable) {
	'-o\0x00\0x00'
};

resource 'DLOG' (256, "Command Set", purgeable) {
	{42, 34, 218, 404},
	dBoxProc,
	invisible,
	noGoAway,
	0x0,
	256,
	""
};

resource 'DITL' (256, "Command Set", purgeable) {
	{	/* array DITLarray: 9 elements */
		/* [1] */
		{24, 296, 48, 352},
		Button {
			enabled,
			"OK"
		},
		/* [2] */
		{64, 296, 88, 352},
		Button {
			enabled,
			"Cancel"
		},
		/* [3] */
		{40, 24, 56, 200},
		RadioButton {
			enabled,
			"Rogue-Like Commands"
		},
		/* [4] */
		{64, 24, 80, 200},
		RadioButton {
			enabled,
			"Original Commands"
		},
		/* [5] */
		{112, 72, 168, 352},
		StaticText {
			disabled,
			"Choose the desired command set.  The change "
			"will take effect when you start the next game."
		},
		/* [6] */
		{16, 16, 96, 232},
		UserItem {
			disabled
		},
		/* [7] */
		{8, 24, 24, 104},
		StaticText {
			enabled,
			"Choose One"
		},
		/* [8] */
		{32, 336, 40, 344},
		UserItem {
			disabled
		},
		/* [9] */
		{112, 16, 144, 48},
		Icon {
			disabled,
			256
		}
	}
};

resource 'CNFG' (257, "TEXT Editor", purgeable) {
	'MACA'
};

resource 'DLOG' (257, "TEXT Editor", purgeable) {
	{42, 34, 226, 426},
	dBoxProc,
	invisible,
	noGoAway,
	0x0,
	257,
	""
};

resource 'DITL' (257, "TEXT Editor", purgeable) {
	{	/* array DITLarray: 11 elements */
		/* [1] */
		{24, 320, 48, 376},
		Button {
			enabled,
			"OK"
		},
		/* [2] */
		{64, 320, 88, 376},
		Button {
			enabled,
			"Cancel"
		},
		/* [3] */
		{40, 24, 56, 200},
		RadioButton {
			enabled,
			"MacWrite"
		},
		/* [4] */
		{64, 24, 80, 176},
		RadioButton {
			enabled,
			"Microsoft Word"
		},
		/* [5] */
		{88, 24, 104, 176},
		RadioButton {
			enabled,
			"Other:  File Creator"
		},
		/* [6] */
		{88, 176, 104, 240},
		EditText {
			disabled,
			""
		},
		/* [7] */
		{136, 72, 176, 376},
		StaticText {
			disabled,
			"Choose what kind of documents you want Moria "
			"to create when it prints to a file."
		},
		/* [8] */
		{16, 8, 120, 264},
		UserItem {
			disabled
		},
		/* [9] */
		{8, 24, 24, 104},
		StaticText {
			enabled,
			"Choose One"
		},
		/* [10] */
		{32, 360, 40, 368},
		UserItem {
			disabled
		},
		/* [11] */
		{136, 16, 168, 48},
		Icon {
			disabled,
			257
		}
	}
};

resource 'DLOG' (258, "Save, Quit, or Cancel", purgeable) {
	{40, 40, 168, 318},
	dBoxProc,
	invisible,
	noGoAway,
	0x0,
	258,
	""
};

resource 'DITL' (258, "Save, Quit, or Cancel", purgeable) {
	{	/* array DITLarray: 6 elements */
		/* [1] */
		{64, 16, 84, 76},
		Button {
			enabled,
			"Save"
		},
		/* [2] */
		{96, 16, 116, 76},
		Button {
			enabled,
			"Quit"
		},
		/* [3] */
		{96, 200, 116, 260},
		Button {
			enabled,
			"Cancel"
		},
		/* [4] */
		{64, 200, 80, 216},
		UserItem {
			disabled
		},
		/* [5] */
		{8, 56, 48, 264},
		StaticText {
			disabled,
			"What do you want to do with the game in progress?"
		},
		/* [6] */
		{8, 16, 40, 48},
		Icon {
			disabled,
			0
		}
	}
};

resource 'MENU' (129, "File") {
	129,
	textMenuProc,
	0x7FFFFFC3,
	enabled,
	"File",
	{	/* array: 7 elements */
		/* [1] */
		"New", noIcon, "N", "", plain,
		/* [2] */
		"Open\311", noIcon, "O", "", plain,
		/* [3] */
		"-", noIcon, "", "", plain,
		/* [4] */
		"Save", noIcon, "S", "", plain,
		/* [5] */
		"Save As\311", noIcon, "", "", plain,
		/* [6] */
		"-", noIcon, "", "", plain,
		/* [7] */
		"Quit", noIcon, "Q", "", plain
	}
};

resource 'MENU' (229, "File (No Keys)") {
	229,
	textMenuProc,
	0x7FFFFFC3,
	enabled,
	"File",
	{	/* array: 7 elements */
		/* [1] */
		"New", noIcon, "", "", plain,
		/* [2] */
		"Open\311", noIcon, "", "", plain,
		/* [3] */
		"-", noIcon, "", "", plain,
		/* [4] */
		"Save", noIcon, "", "", plain,
		/* [5] */
		"Save As\311", noIcon, "", "", plain,
		/* [6] */
		"-", noIcon, "", "", plain,
		/* [7] */
		"Quit", noIcon, "", "", plain
	}
};

resource 'MENU' (133, "Moria") {
	133,
	textMenuProc,
	0x7FFFFF17,
	enabled,
	"Moria",
	{	/* array: 5 elements */
		/* [1] */
		"Help\311", noIcon, "H", "", plain,
		/* [2] */
		"Command Set\311", noIcon, "", "", plain,
		/* [3] */
		"TEXT Editor\311", noIcon, "", "", plain,
		/* [4] */
		"-", noIcon, "", "", plain,
		/* [5] */
		"High Scores\311", noIcon, "", "", plain
	}
};

resource 'MENU' (233, "Moria (No Keys)") {
	233,
	textMenuProc,
	0x7FFFFF17,
	enabled,
	"Moria",
	{	/* array: 5 elements */
		/* [1] */
		"Help\311", noIcon, "", "", plain,
		/* [2] */
		"Command Set\311", noIcon, "", "", plain,
		/* [3] */
		"TEXT Editor\311", noIcon, "", "", plain,
		/* [4] */
		"-", noIcon, "", "", plain,
		/* [5] */
		"High Scores\311", noIcon, "", "", plain
	}
};

resource 'DLOG' (512, "Mac Help", purgeable) {
	{40, 34, 310, 496},
	dBoxProc,
	invisible,
	noGoAway,
	0x0,
	512,
	""
};

resource 'DITL' (512, "Mac Help", purgeable) {
	{	/* array DITLarray: 6 elements */
		/* [1] */
		{16, 392, 36, 452},
		Button {
			enabled,
			"Ok"
		},
		/* [2] */
		{24, 432, 32, 440},
		UserItem {
			disabled
		},
		/* [3] */
		{48, 8, 264, 440},
		UserItem {
			disabled
		},
		/* [4] */
		{48, 439, 264, 455},
		Control {
			enabled,
			512
		},
		/* [5] */
		{8, 52, 40, 380},
		StaticText {
			disabled,
			"Moria for the Macintosh Version 5.2.1"
		},
		/* [6] */
		{8, 8, 40, 40},
		Icon {
			disabled,
			128
		}
	}
};

resource 'CNTL' (512, "Mac Help Scroll Bar", purgeable) {
	{0, 0, 216, 16},
	0,
	visible,
	0,
	0,
	scrollBarProc,
	0,
	""
};

resource 'TEXT' (512, "Mac Help", purgeable) {
	"This is an beta version of Mac Moria, implementation 2.0b1, which is "
	"based on the Umoria 5.2.1 sources.  Please send comments and bug repor"
	"ts to wilson@ernie.Berkeley.EDU (Jim Wilson) or 73230.224@compuserve.c"
	"om (Curtis McCauley).\n\n"
	"Users of previous versions o"
	"f Mac Moria should discard the old program.  Otherwise, the Finder mig"
	"ht get confused when you double-click on a saved game to start Moria.\n"
	"\n*****************************\n\nMoria is a dungeon exploration game.  "
	"The name comes from \322The Lord of the Rings\323 trilogy written by J. R. R"
	". Tolkien.  Moria is pronounced with the accent on the first syllable,"
	" e.g. MORE-ee-ah.  For details on the game itself, please read the Mor"
	"ia Docs file that is distributed along with Moria.\n\nMoria was original"
	"ly written for VAX/VMS systems.  As such,  it is heavily dependent on "
	"the keyboard for game play.  The mouse is not used at all, except for "
	"standard things like menus and dialogs.  To get a list of the availabl"
	"e keyboard commands, hit the \324?\325 key while playing the game.\n\nSome of "
	"the commands are specified as being control characters, i.e.  saving t"
	"he game is ^X.  To type these characters hold down the control key and"
	" then press the appropriate character key.  If you do not have a contr"
	"ol key, use the command key (apple/cloverleaf key) instead of the cont"
	"rol key.  If you prefer, you may enter these commands using two keystr"
	"okes, by first typing the '^' key followed by the letter.\n\nThe game al"
	"so uses the escape key (ESC) quite a bit.  If you do not have an escap"
	"e key on your keyboard, you can use the ` key or the Clear key on the "
	"numeric keypad instead.  An escape can also by typed as ^[, i.e. contr"
	"ol (or command) left bracket.\n\nFile Menu\n\nThe game fully supports the "
	"standard Macintosh file menu.  Use New to start playing a new game,  o"
	"r Open to restart a previous game.   Save As will always ask you for t"
	"he name of the file to save your character in.  Save will usually save"
	" the game in a file with the same name as the name of the current char"
	"acter.  If you are playing a new game, or have changed the name of you"
	"r character, Save will instead ask you for the name of the save file. "
	" The Quit option, of course, is used to exit the game.\n\nA game can onl"
	"y be saved at the point where it is your turn to move.  At all other t"
	"imes the Save command will not be available.\n\nThe File Menu is disable"
	"d while saving and restoring games.  There is no cause for alarm if yo"
	"u see this occur.  After the operation is completed, there will be a p"
	"rompt that says \322press any key to continue\323.  After you press a key, t"
	"he File Menu will be reenabled.\n\nIf your character dies, or you decide"
	" to stop running him, you will be prompted via a standard file dialog "
	"for the name of a file in which to save his memories.  Pressing cancel"
	" will avoid saving the memories.\n\nEdit Menu\n\nThe Edit Menu items are n"
	"ot used by the game.  The Edit menu is disabled unless there is a Desk"
	" Accessory open.\n\nScreen/FontSize Menu\n\nIf you have a color system, th"
	"e Color item allows you to choose the background and foreground colors"
	" of the game window.  The game must use a constant width font, so only"
	" the Monaco font is supported.  You can, however, specify which point "
	"size that you want to use.  The menu will list every point size availa"
	"ble on your Macintosh for the Monaco font.\n\nAdvanced users can choose "
	"a different font by using ResEdit or a similar program to edit the STR"
	" \"Font Name\" resource.\n\nMoria Menu\n\nHelp brings up this dialog.\n\nComma"
	"nd Set allows you to choose between the original Moria command set and"
	" an optional command set similar to the UNIX game Rogue.  The main dif"
	"ference between the two is that the original command set uses the nume"
	"ric keypad for movement, whereas the Rogue-like command set uses the \324"
	"hjkl\325 keys for, respectively, left down up and right.  The program wil"
	"l remember your setting and use it as a default the next time you star"
	"t a new character.  The command set can be chosen for the current char"
	"acter by using the set options ('=') command.\n\nNote that the numeric k"
	"eypad, if your system has one, is always available for indicating dire"
	"ctions, regardless of the command set chosen.  The arrow keys will wor"
	"k as well.\n\nText Editor allows you to choose the type of TEXT file cre"
	"ated by Moria, that is, it allows you to specify which application wil"
	"l be used by the Finder when you open a TEXT file created by Moria.  Y"
	"ou can choose MacWrite or Microsoft Word files.  You can also specify "
	"the signature of the desired application if you prefer something else."
	"\n\nHigh Scores displays all of the scores stored in the scorefile.  You"
	" can clear these scores by just deleting the High Scores file.  This f"
	"eature is not implemented.\n\nConfiguration Info\n\nMoria is fully Multifi"
	"nder compatible/aware.  It requires 450 K of memory under Multifinder."
	"  The game will run on any Macintosh with 1 Megabyte or more of memory"
	".\n\nMoria comes with a documentation file, called Moria Docs.  The Mori"
	"a Docs file is an explanation of the game system.  Please remember to "
	"always distribute this file with the game, since it contains much impo"
	"rtant information.\n\nMoria will create two files when it is run.  The H"
	"igh Scores file keeps a list of the scores of all of the previous game"
	"s.  If this file is not present, a new one will be created in the fold"
	"er which contains the game.  This file can be kept either in the folde"
	"r that contains the application, or in the System Folder.\n\nMoria also "
	"creates a file called Moria Config in the System Folder.  This file co"
	"ntains all of the configuration info for Moria, such as the size and l"
	"ocation of windows, the font size, the command set, etc.  This file mu"
	"st remain in the Sytem Folder.\n\nThe text displayed when Moria starts u"
	"p is stored in the data fork of this program.  You can edit it with mo"
	"st text editors by first changing the file type of the program from 'A"
	"PPL' to 'TEXT' with ResEdit.  Remember to change the file type back ag"
	"ain when you are through.  Be careful not to disturb the lines which b"
	"egin with a '#'.\n\0x00"
};

type 'DFPR' {
	unsigned hex integer = $600A;					/* branch around header */
	unsigned hex integer;							/* flags */
	unsigned literal longint;						/* resource type */
	integer;										/* resource id */
	integer;										/* version */
	unsigned hex integer = $4EF9;					/* jump instruction */
	unsigned hex longint = $0;						/* target address for jump */
};

type 'LDEF' as 'DFPR';

resource 'LDEF' (514, "High Scores", purgeable) {
	$0, 'LDEF', 514, 0
};

resource 'DITL' (514, "High Scores", purgeable) {
	{	/* array DITLarray: 5 elements */
		/* [1] */
		{15, 422, 35, 482},
		Button {
			enabled,
			"Ok"
		},
		/* [2] */
		{24, 464, 32, 472},
		UserItem {
			disabled
		},
		/* [3] */
		{48, 2, 281, 490},
		UserItem {
			disabled
		},
		/* [4] */
		{8, 56, 40, 336},
		StaticText {
			disabled,
			"Moria High Scores\n"
			"Positions ^0 to ^1"
		},
		/* [5] */
		{8, 8, 40, 40},
		Icon {
			disabled,
			128
		}
	}
};

resource 'DLOG' (514, "High Scores", purgeable) {
	{40, 34, 327, 526},
	dBoxProc,
	invisible,
	noGoAway,
	0x0,
	514,
	""
};

resource 'DITL' (1024, "General Error", purgeable) {
	{	/* array DITLarray: 2 elements */
		/* [1] */
		{80, 304, 100, 364},
		Button {
			enabled,
			"Ok"
		},
		/* [2] */
		{7, 68, 71, 364},
		StaticText {
			disabled,
			"^0"
		}
	}
};

resource 'ALRT' (1024, "General Error", purgeable) {
	{42, 42, 148, 416},
	1024,
	{	/* array: 4 elements */
		/* [1] */
		OK, visible, sound1,
		/* [2] */
		OK, visible, sound1,
		/* [3] */
		OK, visible, sound1,
		/* [4] */
		OK, visible, sound1
	}
};

resource 'SIZE' (-1) {
	dontSaveScreen,
	acceptSuspendResumeEvents,
	enableOptionSwitch,
	canBackground,
	multiFinderAware,
	backgroundAndForeground,
	dontGetFrontClicks,
	ignoreChildDiedEvents,
	not32BitCompatible,
	reserved,
	reserved,
	reserved,
	reserved,
	reserved,
	reserved,
	reserved,
	375 * 1024,
	375 * 1024
};
